e2e/cleanup.go

```diff
@@ -0,0 +1,95 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"path"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/filter"
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/meta"
+)
+
+func matchTestResource(name string) bool {
+	if RunID == "" {
+		return strings.HasPrefix(name, TestFlags.ResourcePrefix)
+	} else {
+		return strings.HasPrefix(name, fmt.Sprintf("%s%s-", TestFlags.ResourcePrefix, RunID))
+	}
+}
+
+func cleanupMeshes(ctx context.Context) {
+	tcprs, err := TheCloud.Meshes().List(ctx, filter.None)
+	if err != nil {
+		log.Printf("FallbackCleanup: TheCloud.Meshes().List(ctx, _): %v\n", err)
+		return
+	}
+	for _, tcpr := range tcprs {
+		name := path.Base(tcpr.Name)
+		if !matchTestResource(name) {
+			continue
+		}
+		key := meta.GlobalKey(name)
+		err = TheCloud.Meshes().Delete(ctx, key)
+		log.Printf("FallbackCleanup: TheCloud.Meshes().Delete(ctx, %s): %v\n", key, err)
+	}
+}
+
+func cleanupTcpRoutes(ctx context.Context) {
+	tcprs, err := TheCloud.TcpRoutes().List(ctx, filter.None)
+	if err != nil {
+		log.Printf("FallbackCleanup: TheCloud.TcpRoutes().List(ctx, _): %v\n", err)
+		return
+	}
+	log.Printf("Listed %d tcproutes", len(tcprs))
```

Comment:
nit: this looks like a debug log I think we can get rid of it

Reply:
removed